Common/Statistics/AssetsUnderManagementCapacityManager.cs

```diff
@@ -0,0 +1,238 @@
+
+                    var holdingsTurnover = 1m;
+                    var holdingsValue = security.Holdings.AbsoluteHoldingsValue;
+
+                    if (holdingsValue != 0)
+                    {
+                        var order = _portfolio.Transactions.GetOrderById(orderId);
+                        var orderValue = Math.Abs(order.GetValue(security));
+                        holdingsTurnover = orderValue / holdingsValue;
+
+                        if (holdingsTurnover == 0)
+                        {
+                            RemoveConsolidator(sender, symbol);
+                            return;
+                        }
+                    }
+
+                    var totalMarketVolume = GetTotalMarketVolume(data);
+                    var totalTradeVolumeCapacity = _maximumTradeableVolume * totalMarketVolume / holdingsTurnover;
+
+                    // If it is a new day, we discard the previous day data
+                    var utcDate = data.EndTime.ConvertToUtc(security.Exchange.TimeZone).Date;
+                    if (_previousInputTime != utcDate)
+                    {
+                        _previousInputTime = utcDate;
+                        _maximumSymbolCapacity.Clear();
+                        _historicalPortfolioCapacity.Add(0m);
+                    }
+
+                    // Updates AUM Capacity if there is new information or lower total trade volume capacity for the security
+                    decimal capacity;
+                    if (!_maximumSymbolCapacity.TryGetValue(symbol, out capacity) ||
+                        totalTradeVolumeCapacity < capacity)
+                    {
+                        _maximumSymbolCapacity[symbol] = totalTradeVolumeCapacity;
+                        _historicalPortfolioCapacity[0] = _maximumSymbolCapacity.Sum(x => x.Value);
+                    }
+
+                    RemoveConsolidator(sender, symbol);
+                }
+                catch (Exception e)
+                {
+                    Log.Error($"AssetsUnderManagementCapacityManager.Update: {e}");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the total market volume which is a fraction of the current trade bar volume
+        /// </summary>
+        /// <param name="data">Last price trade bar available or consolidated</param>
+        private static decimal GetTotalMarketVolume(IBaseData data)
+        {
+            var tradeBar = (TradeBar) data;
+
+            var factor = tradeBar.Period == Time.OneDay
+                ? .025m
+                : .050m;
+
+            return factor * tradeBar.Volume;
+        }
+
+        /// <summary>
+        /// Remove consolidator for a given symbol
+        /// </summary>
+        /// <param name="sender">Consolidator to be removed</param>
+        /// <param name="symbol">Symbol associated with the consolidator</param>
+        private void RemoveConsolidator(object sender, Symbol symbol)
```

Comment:
Since this method isn't directly used as an event handler we should remove the `object sender` and replace it with a more accurate type (perhaps `IDataConsolidator`). Also, if I'm reading this correctly, not sure if we'll ever remove consolidators for hour/daily resolutions.